### PR TITLE
feat: add LZ4 compression in storage

### PR DESCRIPTION
**Pull request by aloks98**
I have added the LZ4 package for compression, but I cannot get over the fact that the compressed sizes are coming greater than the actual file sizes. Is there something I am missing? Can somebody please help me with this?

### CODE REVIEW

**Comment by stnguyen90** [comment]
Only local tests:

image

**Comment by stnguyen90** [comment]
> I cannot get over the fact that the compressed sizes are coming greater than the actual file sizes

@aloks98, which files are you referring to?

**Comment by stnguyen90** [comment]
> I cannot get over the fact that the compressed sizes are coming greater than the actual file sizes. Is there something I am missing? Can somebody please help me with this?

@aloks98, compressing doesn't always yield a smaller file, especially really small files, already compressed files, and files with totally random data. I tested your implementation on the RGB 8 bit version of the artificial image [here](https://imagecompression.info/test_images/) and a file compressed to 13% of the original.

**Review by stnguyen90 — APPROVED** [review]

**Comment by stnguyen90** [comment]
@aloks98 Hey 👋 awesome work on your PR! We've approved your work and it'll be merged soon!

File: Dockerfile [hunk 1]
```diff
@@ -18,13 +18,15 @@ RUN composer update \
 FROM php:8.0-cli-alpine as compile
 
 ENV PHP_ZSTD_VERSION="master"
+ENV PHP_LZ4_VERSION="master"
```
Comment (christyjacob4) [inline Dockerfile:21]: Lets not use `master` and stick to using the latest commit hash

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments

**Comment by aloks98** [comment]
hey @christyjacob4
I have updated the version to last commit hash

File: Dockerfile [hunk 2]
```diff
@@ -18,13 +18,15 @@ RUN composer update \
 FROM php:8.0-cli-alpine as compile
 
 ENV PHP_ZSTD_VERSION="master"
+ENV PHP_LZ4_VERSION="8ce521e086fcc4d81c57a60915676673e341ab05"
```
Comment (stnguyen90) [inline Dockerfile:21]: Let's call this COMMIT instead:

```suggestion
ENV PHP_LZ4_COMMIT="8ce521e086fcc4d81c57a60915676673e341ab05"
```

File: Dockerfile [hunk 3]
```diff
@@ -34,20 +36,30 @@ RUN git clone --recursive --depth 1 --branch $PHP_ZSTD_VERSION https://github.co
   && ./configure --with-libzstd \
   && make && make install
 
+## LZ4 Extension
+FROM compile AS lz4
+RUN git clone --recursive --depth 1 --branch $PHP_LZ4_VERSION https://github.com/kjdev/php-ext-lz4.git \
+  && cd php-ext-lz4 \
```
Comment (stnguyen90) [inline Dockerfile:42]: And this is how you check out a specific commit:
```suggestion
RUN git clone --recursive --depth 1 https://github.com/kjdev/php-ext-lz4.git \
  && cd php-ext-lz4 \
  && git reset --hard $PHP_LZ4_COMMIT \
```

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Sorry, would you please make some additional changes? It seems like the image is failing to build.

**Comment by aloks98** [comment]
Alright, got it!

**Review by stnguyen90 — APPROVED** [review]

**Comment by stnguyen90** [comment]
Thank you! 🙏

**Comment by christyjacob4** [comment]
Please sync your branch with `main` . It has fixes for the failing CI tests

**Review by christyjacob4 — APPROVED** [review]

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**